### PR TITLE
Update GitHub CI to use actions/devx

### PR DESCRIPTION
**Pull request by sgillespie**
# Description

Why [input-output-hk/actions/devx-action](https://github.com/input-output-hk/actions#devx-action)? The devx action provides all the system dependencies with nix, but haskell dependencies from cabal. This makes builds easily reproducible from both CI and local development.

This means that if anything goes wrong in CI, you can enter the devx shell, then run the same commands CI runs, and the result should be the same.

# Checklist

- [x] Commit sequence broadly makes sense
- [x] Commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] Any changes are noted in the [changelog](https://github.com/input-output-hk/cardano-db-sync/blob/master/cardano-db-sync/CHANGELOG.md)
- [ ] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) on version 0.10.1.0 (which can be run with `scripts/fourmolize.sh`)
- [x] Self-reviewed the diff

# Migrations

- [ ] The pr causes a [breaking change](https://github.com/input-output-hk/cardano-db-sync/blob/master/doc/migrations.md) of type a,b or c
- [ ] If there is a breaking change, the pr includes a database migration and/or a fix process for old values, so that upgrade is possible
- [ ] Resyncing and running the migrations provided will result in the same database semantically

If there is a breaking change, especially a big one, please add a justification here. Please elaborate
more what the migration achieves, what it cannot achieve or why a migration is not possible.

### CODE REVIEW

**Review by Cmdv — COMMENTED** [review]
LGTM 🎉 
one little question will I still be ok to run ghc 9.6.2 locally?

**Review by Cmdv — APPROVED** [review]

**Comment by sgillespie** [comment]
> LGTM 🎉 one little question will I still be ok to run ghc 9.6.2 locally?

Good point, I need to add this back to the matrix

**Review by kderme — APPROVED** [review]

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -1,143 +1,82 @@
 name: Haskell CI
 
-on: [push]
+on:
+  push:
+  pull_request:
+  schedule:
+    # Everyday at 4:00 AM
```
Comment (angerman) [inline .github/workflows/haskell.yml:7]: Do we know the TZ? is this UTC?

File: .github/workflows/haskell.yml [hunk 2]
```diff
@@ -1,143 +1,82 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Install GHC and Cabal
+        uses: input-output-hk/actions/devx@latest
+        with:
+          platform: ${{ matrix.os == 'ubuntu-latest' && 'x86_64-linux' || 'x86_64-darwin' }}
+          target-platform: ""
+          compiler-nix-name: ${{ matrix.compiler-nix-name }}
+          minimal: false
+          iog: true
```
Comment (angerman) [inline .github/workflows/haskell.yml:34]:
```suggestion
          # enable IOG flavour to bring in all the crypto libraries we need.
          iog: true
```

**Review by angerman — APPROVED** [review]
Just a few ideas. Should also `shellcheck` `scripts/postgresql-test.sh` in ci?

File: .github/workflows/haskell.yml (hunk 1 above)
Comment (sgillespie) [reply .github/workflows/haskell.yml:7]: Great question! It appears to be [UTC](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onschedule). Is there a better time?

**Comment by sgillespie** [comment]
> Just a few ideas. Should also `shellcheck` `scripts/postgresql-test.sh` in ci?

There are a lot of violations in all of our other shell scripts, so I'll address that in another PR. In the meantime, I've fixed all the violations in `postgres-test.sh`

File: .github/workflows/haskell.yml (hunk 1 above)
Comment (angerman) [reply .github/workflows/haskell.yml]: Let's just document that it's 4am utc :-) 4 am is such an ambiguous value without a TZ.

Comment (sgillespie) [reply .github/workflows/haskell.yml]: Done

**Comment by sgillespie** [comment]
> Should also `shellcheck` `scripts/postgresql-test.sh` in ci?

https://github.com/input-output-hk/cardano-db-sync/pull/1508

**Review by Cmdv — APPROVED** [review]

**Review by angerman — APPROVED** [review]
Thank you! This looks good :D So much deleted code 🥳

**Pull request merged**